velox/exec/tests/utils/QueryAssertions.cpp

```diff
@@ -754,15 +764,7 @@ std::shared_ptr<Task> assertQuery(
   }
   auto task = cursor->task();
 
-  if (not task->isFinished()) {
-    // The Task can return results before the Driver is finished executing.
-    // Wait for the Task to finish before returning it to ensure it's stable
-    // e.g. the Driver isn't updating it anymore.
-    auto& executor = folly::QueuedImmediateExecutor::instance();
-    auto future = task->stateChangeFuture(1'000'000).via(&executor);
-    future.wait();
-    EXPECT_TRUE(task->isFinished());
-  }
+  waitForTaskCompletion(task.get());
```

Comment:
You need to wrap this in EXPECT_TRUE to preserve the original behavior.

Reply:
good catch!